### PR TITLE
Make adminlog path customizable

### PR DESCRIPTION
**Pull request by anlutro**

### CODE REVIEW

**Comment by teward** [comment]
Have you tested this to verify that you have configuration capabilities added to support this?  I.E. this path is configurable by direct message or argument loadtime, as well as stored somewhere in the config?  (I haven't tested, but these're valid questions)

File: modules/adminlog.cpp [hunk 1]
```diff
@@ -46,7 +46,11 @@ class CAdminLogMod : public CModule {
 		else
 			m_eLogMode = LOG_TO_FILE;
 
-		m_sLogFile = GetSavePath() + "/znc.log";
+		CString sPath = GetNV("path");
+		if (sPath.Equals(""))
+			m_sLogFile = GetSavePath() + "/znc.log";
+		else
+			m_sLogFile = sPath
```
Comment (teward) [inline modules/adminlog.cpp]: Missing a semicolon at the end here, so it'll Fail To Build.

**Comment by anlutro** [comment]
That didn't occur to me as I manage my registry files manually. Thanks for the reminder! I'll implement a command for setting the path.

File: modules/adminlog.cpp [hunk 2]
```diff
@@ -85,6 +87,30 @@ class CAdminLogMod : public CModule {
...
+
+		CFile LogFile(sPath);
+		CString sLogDir = LogFile.GetDir();
+		struct stat ModDirInfo;
+		CFile::GetInfo(GetSavePath(), ModDirInfo);
+		if (!CFile::Exists(sLogDir)) {
+			CDir::MakeDir(sLogDir, ModDirInfo.st_mode);
+		}
+
+		if (!sPath.empty()) {
```
Comment (jpnurmi) [inline modules/adminlog.cpp]: sPath is never empty at this point?

File: modules/adminlog.cpp [hunk 3]
```diff
@@ -85,6 +87,30 @@ class CAdminLogMod : public CModule {
...
+			m_sLogFile = sPath;
+			SetNV("path", sPath);
+			sMessage = "adminlog file path set to [" + sPath + "]";
+			Log(sMessage);
+			PutModule(sMessage);
```
Comment (jpnurmi) [inline modules/adminlog.cpp]: it probably shouldn't do these when called from OnLoad()

File: modules/adminlog.cpp [hunk 4]
```diff
@@ -27,6 +27,7 @@ class CAdminLogMod : public CModule {
 		AddHelpCommand();
 		AddCommand("Show", static_cast<CModCommand::ModCmdFunc>(&CAdminLogMod::OnShowCommand), "", "Show the logging target");
 		AddCommand("Target", static_cast<CModCommand::ModCmdFunc>(&CAdminLogMod::OnTargetCommand), "<file|syslog|both>", "Set the logging target");
+		AddCommand("Path", static_cast<CModCommand::ModCmdFunc>(&CAdminLogMod::OnPathCommand), "<path>", "If using the file target, set the log file path");
```
Comment (jpnurmi) [inline modules/adminlog.cpp]: just to throw an idea on the table... perhaps the path could be passed as an optional argument to the target command? not sure which is better.

**Comment by anlutro** [comment]
Thanks for the feedback @jpnurmi - I agree with your suggestion of making the path a second arg to the target command. I suppose the main disadvantage is you can't switch between "file" and "both" without specifying the path over again.

File: modules/adminlog.cpp [hunk 5]
```diff
@@ -85,6 +86,23 @@ class CAdminLogMod : public CModule {
 		Log("[" + sUsername + "] failed to login from " + sRemoteIP, LOG_WARNING);
 	}
 
+	void SetLogFilePath(CString& sPath) {
```
Comment (DarthGandalf) [inline modules/adminlog.cpp]: const CString&

Comment (DarthGandalf) [inline modules/adminlog.cpp]: Or, just CString. As it's modified below.
The point is, it shouldn't be modified on caller side.

File: modules/adminlog.cpp [hunk 6]
```diff
@@ -46,7 +46,8 @@ class CAdminLogMod : public CModule {
 		else
 			m_eLogMode = LOG_TO_FILE;
 
-		m_sLogFile = GetSavePath() + "/znc.log";
+		CString sPath = GetNV("path");
+		SetLogFilePath(sPath);
```
Comment (DarthGandalf) [inline modules/adminlog.cpp]: After the fix below, `SetLogFilePath(GetNV("path"))` would be enough

**Comment by anlutro** [comment]
@DarthGandalf done

**Pull request closed without merging**